File: app/src/main/java/com/tosh/poolandroid/View/PhoneActivity.java
package com.tosh.poolandroid.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.material.button.MaterialButton;

import com.tosh.poolandroid.R;
import com.tosh.poolandroid.Remote.AuthRetrofitClient;
import com.tosh.poolandroid.Remote.NodeAuthService;
import com.tosh.poolandroid.ViewModel.PhoneViewModel;

import io.reactivex.disposables.CompositeDisposable;
import retrofit2.Retrofit;

public class PhoneActivity extends AppCompatActivity {

    private MaterialButton proceedBtn;
    private EditText inputPhone;

    private String email;
    private String name;

    private SharedPreferences pref;
    private PhoneViewModel phoneViewModel;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_phone);

        pref= PreferenceManager.getDefaultSharedPreferences(getBaseContext());
        email=pref.getString("email", "default");
        name = pref.getString("name", "default");

        //init views
        proceedBtn = findViewById(R.id.proceed_btn);
        inputPhone = findViewById(R.id.phone_input);

        proceedBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getPhone(name, inputPhone.getText().toString(), email);
            }
        });

        instatiatePhoneViewModel();


    }

    private void getPhone(String name, String phone, String email) {
        if(TextUtils.isEmpty(phone)){
            Toast.makeText(this, "Enter phone", Toast.LENGTH_SHORT).show();
            return;
        }
        if(phone.length()!= 9){
            Toast.makeText(this, "Enter valid phone number", Toast.LENGTH_SHORT).show();
            return;
        }

        phoneViewModel.getPhone(name,phone,email);
    }
    private void instatiatePhoneViewModel(){
        phoneViewModel = ViewModelProviders.of(this).get(PhoneViewModel.class);

        phoneViewModel.getPhoneStatus().observe(this, new Observer<String>() {
            @Override
            public void onChanged(String s) {
                if (s.equals("success")){
                    Intent intent = new Intent(PhoneActivity.this, MainActivity.class);
                            startActivity(intent);
                            finish();
                }else{
                    Toast.makeText(PhoneActivity.this, s, Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: app/src/main/java/com/tosh/poolandroid/view/activity/PhoneActivity.kt
package com.tosh.poolandroid.view.activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.animation.DecelerateInterpolator
import android.view.animation.Interpolator
import android.widget.LinearLayout
import android.widget.Toast
import androidx.core.content.ContextCompat
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.agrawalsuneet.dotsloader.loaders.LazyLoader
import com.tosh.poolandroid.R
import com.tosh.poolandroid.util.show
import com.tosh.poolandroid.viewmodel.MainViewModel
import kotlinx.android.synthetic.main.activity_phone.*

class PhoneActivity : AppCompatActivity() {
    lateinit var containerLL: LinearLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_phone)

        containerLL = findViewById<LinearLayout>(R.id.containerLL)
        val intent = intent
        var email:String = intent.getStringExtra("Email")
        var name:String = intent.getStringExtra("Name")

        proceed_btn.setOnClickListener {
            val phone = phone_input.text.toString().trim()

            if (phone.isEmpty()){
                phone_input.error = "Enter a phone number"
                Toast.makeText(this, "Enter a phone number", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (phone.length !=9){
                phone_input.error = "Enter a valid number"
                Toast.makeText(this, "Enter a valid number", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            loading()
            instantiatePhoneViewModel(name, email,phone)
        }
    }

    private fun instantiatePhoneViewModel(name:String, email:String, phone:String) {
        containerLL.visibility = View.VISIBLE
        val phoneViewModel: MainViewModel = ViewModelProviders.of(this)[MainViewModel::class.java]
        phoneViewModel.addUserPhone(name, email, phone).observe(this, Observer {
            if (it == "successful"){
                val intent = Intent(applicationContext, MainActivity::class.java)
                startActivity(intent)
                finish()
            }else{
                Toast.makeText(applicationContext,it, Toast.LENGTH_LONG).show()
                containerLL.visibility = View.GONE
            }
        })

    }

    private fun loading(){
        var lazyLoader = LazyLoader(this, 15, 5,
            ContextCompat.getColor(this, R.color.loader_selected),
            ContextCompat.getColor(this, R.color.loader_selected),
            ContextCompat.getColor(this, R.color.loader_selected))
            .apply {
                animDuration = 500
                firstDelayDuration = 100
                secondDelayDuration = 200
                interpolator = DecelerateInterpolator() as Interpolator
            }
        containerLL.addView(lazyLoader)
    }
}
